src/clj/game/cards/hardware.clj

```diff
@@ -9,7 +9,7 @@
     {:successful-run
      {:silent (req true)
       :delayed-completion true
-      :req (req (= target :archives))
+      :req (req (and (= target :archives) (not-empty (:discard corp))))
```

Comment:
Looks good. Clojure people would prefer using the `seq` function to test for not-emptiness, as using `seq` on something that is empty will return `nil`. I think that's less readable, and prefer `not-empty`.